tests/test_settings.py

```diff
@@ -1,24 +1,51 @@
 # -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
 import pytest
 
 
 @pytest.mark.usefixtures('coveragerc')
-def test_project_currencies(testdir):
-    testdir.makepyfile(test_settings='''
-    INSTALLED_APPS = ['djmoney']
-    CURRENCIES = ['USD', 'EUR']
-    SECRET_KEY = 'foobar'
-    ''')
-    testdir.makepyfile('''
-    def test_project_currencies():
-        from djmoney.settings import CURRENCY_CHOICES
-
-        assert CURRENCY_CHOICES == [('EUR', 'Euro'), ('USD', 'US Dollar')]
-    ''')
-    result = testdir.runpytest_subprocess(
-        '--verbose', '-s',
-        '--ds', 'test_settings',
-        '--cov', 'djmoney',
-        '--cov-config', 'coveragerc.ini',
-    )
-    assert 'test_project_currencies.py::test_project_currencies PASSED' in result.stdout.lines
+class TestCurrencies:
+
+    def assert_test_passed(self, testdir, lines):
+        result = testdir.runpytest_subprocess(
+            '--verbose', '-s',
+            '--ds', 'app_settings',
+            '--cov', 'djmoney',
+            '--cov-config', 'coveragerc.ini',
+        )
+        result.stdout.fnmatch_lines(lines)
```

Comment:
1. No, it runs only tests defined in the temp directory, in these cases it is 1 subprocess test per test method in the class
2. If we load ``djmoney.settings`` in first test once it will load Django's settings. In second test if we need to test another combination of settings we will need to change them in settings.py, remove `djmoney/settings.pyc` and load `djmoney.settings` again to get new values and cover all logic branches in `djmoney.settings`. I'm not sure if we can do it in the same way on all Django/Python combinations that we support. So, this solution is rough but isolated :)
3. Almost, in this case the most problematic thing is that avoiding module reloading and caching issues seems to me core complicated than running every test in the isolated environment.

I'm not sure about [https://docs.djangoproject.com/en/dev/topics/testing/tools/#overriding-settings](https://docs.djangoproject.com/en/dev/topics/testing/tools/#overriding-settings), I'll try it :)

Probably running python with -B option could help. I will try to simplify tests :)